lib/rspec/matchers/built_in/include.rb

```diff
@@ -107,8 +107,10 @@ def comparing_hash_to_a_subset?(expected_item)
         end
 
         def actual_hash_includes?(expected_key, expected_value)
-          actual_value = actual.fetch(expected_key) { return false }
-          values_match?(expected_value, actual_value)
+          actual.any? do |actual_key, actual_value|
+            values_match?(expected_key, actual_key) &&
+              values_match?(expected_value, actual_value)
+          end
```

Comment:
So... this changes the matcher to iterate over the hash, which could have an impact on performance in large hashes... Something like this:

```Ruby
actual_value =
  actual.fetch(expected_key) do
    actual.find(proc { return false }) { |actual_key, _| values_match?(expected_key, actual_key) }[0]
  end
values_match?(expected_value, actual_value)
```

Would allow shortcuts for set keys, but still search for fuzzy keys... (The extra proc keeps the `return false` safeguard and prevents `if`s)... Its not as pretty... but shouldn't cause regressions in performance.... I'm on the fence as to wether thats better...

Reply:
Nice, I love that much more than my solution.

Do you want me to benchmark on huge large hashes? I believe that with normal keys which are significantly more common it will sure be faster.

It also doesn't break `have_broadcasted_to` :D

Small correction `[0]` => `[1]`.

Reply:
Er... Why `1`? Oh because you want the value not the key?

Reply:
Yep 👍